src/App.js

```diff
@@ -4,7 +4,32 @@ import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export class App extends Component {
-  state = {};
+  state = {
+    movies: [...moviesFromServer],
+    query: '',
+  };
+
+  handleChange = (event) => {
+    this.setState({
+      query: (event.target.value).toLowerCase(),
+      movies: [...moviesFromServer],
+    });
+    this.moviesFilter();
```

Comment:
since you do filtering here, use the value you got from the input, not the state, because setState set the value asynchronously and you get it with delay